backend/src/packages/auth/auth.service.ts

```diff
@@ -75,8 +75,11 @@ class AuthService {
     };
   }
 
-  public getAuthenticatedUser(id: number): Promise<UserAuthResponseDto | null> {
-    return this.userService.findById(id);
+  public async getAuthenticatedUser(id: number): Promise<UserAuthResponseDto> {
+    const user = (await this.userService.findById(id)) as UserAuthResponseDto;
```

Comment:
```suggestion
    const user = (await this.userService.findById(id)) as UserAuthResponseDto;
```
What if this.userService.findById(id) returns null? Will it work as expected?

Reply:
I think yes. To get this response we first need to sign in. If sign-in is successful we will have this user in the db. If the opposite, then it would be the error about a nonauthorized user. So in this case `this.userService.findById(id)` wouldn't return null.

Reply:
@HIMDanny Good explanation!

Reply:
this relates on business logic too much, it's better not to expect only user here, but null also

Reply:
ok, then I'll change it